istio-2-mixin/panels.libsonnet

```diff
@@ -0,0 +1,1061 @@
+local g = import './g.libsonnet';
+local commonlib = import 'common-lib/common/main.libsonnet';
+local utils = commonlib.utils;
+{
+  new(this):
+    {
+      local t = this.grafana.targets,
+      local alertList = g.panel.alertList,
+      local stat = g.panel.stat,
+      local timeSeries = g.panel.timeSeries,
+      local pieChart = g.panel.pieChart,
+      local barGauge = g.panel.barGauge,
+      local table = g.panel.table,
+      local histogram = g.panel.histogram,
+
+      alertsPanel: 
+        alertList.new('Istio alerts')
+        + alertList.options.UnifiedAlertListOptions.withAlertInstanceLabelFilter(this.grafana.variables.queriesGroupSelectorAdvanced),
+
+      proxies:
+        commonlib.panels.generic.stat.base.new(
+          'Proxies',
+          targets=[t.proxyCount],
+          description='Number of proxies in the Istio system.'
+        )
+        + stat.options.withGraphMode('none')
+        + stat.standardOptions.color.withMode('thresholds')
+        + stat.standardOptions.thresholds.withSteps([
+          stat.thresholdStep.withColor('super-light-red')
```

Comment:
In the old dashboard, there was an additional mode set to `absolute` instead of `threshold` like:
"thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "super-light-red",
                "value": null
              },
              {
                "color": "super-light-green",
                "value": 1
              }
            ]
          },
          
          Not sure if this needs to be updated.

Reply:
@JonathanWamsley It actually looks like the generated json has mode as "absolute" currently

Reply:
I just generated it and got it showing mode as thresholds while the old json showed it as absolute before. This is because of `+ stat.standardOptions.color.withMode('thresholds')` line. Just want to make sure this is correct:
```json
            "fieldConfig": {
               "defaults": {
                  "color": {
                     "mode": "thresholds"
                  },
                  "thresholds": {
                     "steps": [
                        {
                           "color": "super-light-red",
                           "value": null
                        },
                        {
                           "color": "super-light-green",
                           "value": 1
                        }
                     ]
                  }
               }
            },
```

Reply:
@JonathanWamsley take a look at the JSON in the dashboards in the cloud. I'm seeing this being generated.
```
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "super-light-orange",
                "value": null
              },
              {
                "color": "super-light-green",
                "value": 1
              }
            ]
          }
        },
```

Not sure why you're seeing something different generated. Can you tell me which panel you're looking at specifically?

Reply:
ahh, I see. So the color mode is set to threshold which matches with the old dashboard. Then there is another mode under the `thresholds` that is left empty. But by default, when you generated the dashboard, it populates with absolute which matches with the old dashboard. 👍